clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -383,6 +384,13 @@ synchronized void maybeSeekUnvalidated(TopicPartition tp, long offset, OffsetRes
         return new HashSet<>(this.assignment.partitionSet());
     }
 
+    /**
+     * @return a modifiable copy of the currently assigned partitions as a List
+     */
+    public synchronized List<TopicPartition> assignedPartitionList() {
```

Comment:
I also added a method to SubscriptionState which returns a copy of the assignment as a List, so we can avoid the awkwardness of copying it into a HashSet and then copying that result into an ArrayList

Reply:
SGTM.